### DISCUSSION TITLE
Why filters are not supported in firefox?

### DISCUSSION THREAD

**Discussion by PabloRuizCuevas** [skeletonlabs/skeleton#1180] [Q&A]
For what I see in the dock the img filters has _none support in firefox_, and  it says:

_This feature is enabled by the use of native browser [SVG filters](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/filter) via [feColorMatrix](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feColorMatrix) transformations._

but in for what I see firefox does support  feColorMatrix https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feColorMatrix  and also the page https://fecolormatrix.com/ works perfectly in firefox.

Just wanted to ask if I'm missing any obvious detail about this? I would´t mind to help solving or testing this issue if there is no bigger issues behind.

**Comment by Sarenor** [comment]
Do the Skeleton Filters work for you in Firefox? We may just have missed the update and need to update the documentation then (which you're welcome to do, if you want to!) :)

**Reply by Sarenor** [reply]
It's disabled explicitly in https://github.com/skeletonlabs/skeleton/blob/master/src/lib/actions/Filters/filter.ts#L5
Let me quickly remove that and check it locally.

**Reply by Sarenor** [reply]
Even with that line removed it doesn't work in my Firefox installation. Might be that it's out of date or there might be more debugging required.
I think I remember @endigo9740 disabling this on purpose in Firefox because it wasn't working as expected, but I can't remember what the exact issue was, I'm sorry.

**Reply by PabloRuizCuevas** [reply]
Not "skeleton" ones, but the tool  https://fecolormatrix.com/  does work and applies the filters correctly, so I see no reason why they shouldn't work Firefox, given that the docs says that is using that technology.

**Comment by endigo9740** [comment]
@PabloRuizCuevas MDN isn't quite accurate here, the filters do not work. Given this we explicitly disable them to prevent extra processing for something that doesn't display. If ever Firefox does support them, we'll gladly enable them.

**Reply by PabloRuizCuevas** [reply]
Ok, thanks, just find weird that that the tool of the docs does work in firefox:
![image](https://user-images.githubusercontent.com/48098178/226143779-d610abb1-9c0f-42b6-9682-71dcca5d6d36.png)

In any case I was testing the code in MDN and it seems to work for me, not sure exactly at where is broken, and if the case, would be nice to know if this is a know error by the people of mozilla.
```
<style>
	img {
	  filter: url(#dabadee);
	}
</style>

<img src="testjpg"/>

<svg xmlns="http://www.w3.org/2000/svg">
	<filter id="dabadee">
	<feColorMatrix 
		type="matrix" 
		values="
		.5 .2 .3 0 0
		.1 .5 .1 0 0
		.1 .1 .8 0 0
		0 0 0 1 0">
	</feColorMatrix>
	</filter>
</svg>
```

**Reply by endigo9740** [reply]
@PabloRuizCuevas quick reminder that Skeleton is open source. Feel free to pull down the source and test without the Firefox block in place. Our filters specifically have never worked in Firefox. I've tested as recently as a couple weeks ago.

It could be limited to certain features, certain combinations of values in the matrix, etc. If you can find a fix, send a PR. We'll gladly welcome it.
